### PR TITLE
pkg/parcacol: Correctly handle delta non cpu/ns profiles

### PR DESCRIPTION
**Pull request by metalmatze**
Since it's possible to have delta profiles that aren't cpu/nanoseconds we still need to query them correctly.
Additionally, setting the resultType is fixed again too.

### CODE REVIEW

File: pkg/parcacol/querier.go [hunk 1]
```diff
@@ -393,7 +393,7 @@ func (q *Querier) queryRangeDelta(
 		logicalplan.Col(profile.ColumnDuration),
 	}
 
-	if isSamplesCount(m.SampleType) {
+	if isSamplesCount(resultType) {
```
Comment (brancz) [inline pkg/parcacol/querier.go:396]: I would still use `m.SampleType`. It's no different, and using `resultType` only happens to be the same at this point in time, as `m.SampleType` is the "default".

File: pkg/parcacol/querier.go [hunk 2]
```diff
@@ -1427,10 +1448,15 @@ func (q *Querier) selectMerge(
 	}
 
 	queryParts.Meta.Timestamp = start
+	queryParts.Meta.SampleType = resultType
```
Comment (brancz) [inline pkg/parcacol/querier.go:1451]: I'm not a fan of overriding a sub-struct of queryParts like this, we don't use anything else of queryParts

File: pkg/parcacol/querier.go [hunk 3]
```diff
@@ -412,17 +412,35 @@ func (q *Querier) queryRangeDelta(
...
 
-	perSecondExpr := logicalplan.Div(
-		logicalplan.Convert(totalSum, arrow.PrimitiveTypes.Float64),
-		logicalplan.Convert(
-			logicalplan.If(
-				logicalplan.IsNull(timestampUnique),
-				logicalplan.Literal(step.Nanoseconds()),
-				durationMin,
+	var perSecondExpr logicalplan.Expr
+	if isCPUNanoseconds(resultType) {
```
Comment (brancz) [inline pkg/parcacol/querier.go:416]: just `isNanoseconds` should be sufficient here

File: pkg/parcacol/querier.go (hunk 2 above)
Comment (metalmatze) [reply pkg/parcacol/querier.go:1451]: We return queryParts because those are now passed into SymbolizeArrowRecord to check for delta profiles.

**Pull request merged**